CMakeLists.txt

```diff
@@ -105,13 +80,49 @@ write_basic_package_version_file(
 
 # Use variables:
 #   * TARGETS_EXPORT_NAME
-#   * PROJECT_NAME
+#   * CMAKE_PROJECT_NAME
 configure_package_config_file(
     "cmake/Config.cmake.in"
     "${PROJECT_CONFIG}"
     INSTALL_DESTINATION "${CONFIG_INSTALL_DIR}"
 )
 
+# zip
+set(Header_Files "src/miniz.h" "src/zip.h")
+source_group("Header Files" FILES "${Header_Files}")
+
+set(Source_Files "src/zip.c")
+source_group("Source Files" FILES "${Source_Files}")
+
+add_library(${CMAKE_PROJECT_NAME} "${Header_Files}" "${Source_Files}")
```

Comment:
No, this does not make sense.
`CMAKE_PROJECT_NAME` is the top-level project name. In projects with dependencies as sub-projects, this pattern would give all libraries the same name.